### PR TITLE
[AMDGPU] Add a new target for gfx1153

### PR DESCRIPTION
**Pull request by perlfu**

### CODE REVIEW

**Review by jayfoad — APPROVED** [review]
LGTM. The compiler currently treats it as identical to gfx1152, right?

**Comment by perlfu** [comment]
> The compiler currently treats it as identical to gfx1152, right?

Correct.

File: offload/DeviceRTL/CMakeLists.txt [hunk 1]
```diff
@@ -46,7 +46,7 @@ set(all_amdgpu_architectures "gfx700;gfx701;gfx801;gfx803;gfx900;gfx902;gfx906"
                              "gfx908;gfx90a;gfx90c;gfx940;gfx941;gfx942;gfx1010"
                              "gfx1030;gfx1031;gfx1032;gfx1033;gfx1034;gfx1035"
                              "gfx1036;gfx1100;gfx1101;gfx1102;gfx1103;gfx1150"
-                             "gfx1151;gfx1152")
+                             "gfx1151;gfx1152;gfx1153")
```
Comment (shiltian) [inline offload/DeviceRTL/CMakeLists.txt:49]: Not sure what happened to the indentations when adding the previous gfx11 targets…😂

**Review by shiltian — APPROVED** [review]

File: offload/DeviceRTL/CMakeLists.txt (hunk 1 above)
Comment (jhuber6) [reply offload/DeviceRTL/CMakeLists.txt:49]: We really need to stop doing this "Compile the same runtime 50 times" thing.

File: llvm/docs/AMDGPUUsage.rst [hunk 2]
```diff
@@ -603,26 +610,30 @@ Generic processor code objects are versioned. See :ref:`amdgpu-generic-processor
...
                                          - ``gfx1151``
-                                         - ``gfx1152``                                          Not all VGPRs can be used on:
+                                         - ``gfx1152``
+                                         - ``gfx1153``                                          Not all VGPRs can be used on:
 
                                                                                                 - ``gfx1100``
                                                                                                 - ``gfx1101``
                                                                                                 - ``gfx1151``
                                                                                                 - ``gfx1152``
+                                                                                                - ``gfx1153``
```
Comment (jayfoad) [inline llvm/docs/AMDGPUUsage.rst:620]: Remove this line. This was already wrong for gfx1152, see #113163

**Pull request merged**